narwhal/executor/src/subscriber.rs

```diff
@@ -174,7 +174,8 @@ impl<Network: SubscriberNetwork> Fetcher<Network> {
     ) -> Vec<impl Future<Output = (BatchIndex, Batch)> + '_> {
         self.metrics.subscriber_processed_certificates.inc();
         debug!("Fetching payload for {:?}", deliver);
-        let mut ret = vec![];
+        let mut ret = Vec::with_capacity(deliver.certificate.header.payload.len());
+        let deliver = Arc::new(deliver);
```

Comment:
Although I see the point of this (avoid cloning the output) , but how much of a problem is it? I mean personally I would prefer have this as is and just clone instead of adding another layer on top (Arc) unless we do have evidence that this is an issue.